### PR TITLE
lisav2: move Run-LISAv2.ps1 to a module & add unit tests

### PR DESCRIPTION
**Pull request by ader1990**
To enable unit testing for all the framework I had to move all the code into a module -- LISAv2-Framework.psm1.
The move to a properly defined PowerShell module (using a psd1), will make it easy to add lisav2 to Powershell Gallery so that users can install the module by just running "Install-Module LISAv2-Framework".

To properly define the lisav2 module interface, I created a psd1 file that contains the module description.
In the future, we can add more methods to that module (for example the utilities methods).

This patch is mostly a refactor, and no backward incompatible changes are made to the original functionality.

This PR addresses issue: https://github.com/LIS/LISAv2/issues/603.

Execution tested on HyperV and Azure.

```
[LISAv2 Test Results Summary]
Test Run On           : 01/15/2019 11:04:19
VHD Under Test        : F:\vms\Virtual Hard Disks\ubuntu-1604.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:2

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-CORE-VERIFY-LIS-MODULES                                                       PASS                 0.99
```

### CODE REVIEW

File: Libraries/TestLogs.psm1 [hunk 1]
```diff
@@ -37,14 +41,15 @@ Function Write-Log()
...
+	$logType = $logLevel.PadRight(4, ' ')
+	$finalMessage = "[$now] [$logType] $text"
 	$fgColor = "White"
 	switch ($logLevel)
 	{
 		"INFO"	{$fgColor = "White"; continue}
-		"WARN"	{$fgColor = "Yellow"; continue}
+		"WARN"	{$fgColor = "DarkYellow"; continue}
 		"ERROR"	{$fgColor = "Red"; continue}
+		"VERBOSE"	{$fgColor = "Yellow"; continue}
```
Comment (lizzha) [inline Libraries/TestLogs.psm1]: tab before {

File: TestControllers/TestController.psm1 [hunk 2]
```diff
@@ -24,9 +24,14 @@
 
 #>
 ###############################################################################################
+
 using Module "..\TestProviders\TestProvider.psm1"
 using Module "..\Libraries\TestReport.psm1"
 
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+Import-Module (Join-Path $here "..\Libraries\TestReport") -Force -DisableNameChecking
```
Comment (lizzha) [inline TestControllers/TestController.psm1]: it's not necessary, as TestReport only contains classes, it's imported with "using" statement

File: Libraries/TestReport.psm1 [hunk 3]
```diff
@@ -375,7 +375,11 @@ Class TestSummary
 		$strHtml += $this.HtmlSummary
 		$strHtml += "</table></body></Html>"
 
-		Set-Content -Value $strHtml -Path $FilePath -Force | Out-Null
+		if (Test-Path (Split-Path $FilePath)) {
+			Set-Content -Value $strHtml -Path $FilePath -Force | Out-Null
```
Comment (lizzha) [inline Libraries/TestReport.psm1]: it can be saved successfully even if the file doesn't exist, so it's not necessary to test-path

**Comment by lizzha** [comment]
The PR includes change for enable verbose log, move logic in run-lisav2.ps1 to module, add unit test. Better to rename the title to reflect the changes.
Also please run regression test on the test

**Comment by LiliDeng** [comment]
By default the verbose is turn on?

**Comment by LiliDeng** [comment]
Run  BVT-NET-IFUP-IFDOWN against hyperv, hit below exception 
[01/15/2019 06:50:10] [VERBOSE] echo '******' | sudo -S -s eval "export HOME=`pwd`;bash BVT-NET-IFUP-IFDOWN.sh > BVT-NET-IFUP-IFDOWN.log" is running in background with ID 375 ...
[01/15/2019 06:50:11] [ERROR] EXCEPTION: The term 'Get-IPv4AndWaitForSSHStart' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again BVT-NET-IFUP-IFDOWN

**Comment by ader1990** [comment]
@lizzha @LiliDeng I will remove the code that is not related to Run-LISAv2.ps1 refactor, and focus on this PR only on the module creation and unit testing.

File: Libraries/TestReport.psm1 (hunk 3 above)
Comment (ader1990) [reply Libraries/TestReport.psm1]: I am checking for the parent path if it exists (I have to for the unit tests to pass)

**Comment by LiliDeng** [comment]
Please provide result for BVT-NET-IFUP-IFDOWN  @asofron

**Comment by ader1990** [comment]
> Please provide result for BVT-NET-IFUP-IFDOWN @asofron

```
Test Run On           : 01/16/2019 06:13:47
[azure] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[azure] Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
[azure] Total Time (dd:hh:mm) : 0:0:6
[azure] 
[azure]    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
[azure] ---------------------------------------------------------------------------------------------------------------------
[azure]     1 BVT-NET-IFUP-IFDOWN                                                               PASS                 4.38 


```

**Comment by ader1990** [comment]
@LiliDeng BVT-NET-IFUP-IFDOWN against hyperv fails, but not because of this patch. this branch is not rebased against your fix https://github.com/LIS/LISAv2/pull/606

**Comment by LiliDeng** [comment]
got it

**Comment by ader1990** [comment]
> got it

I rebased on top of master and hyper-v passes too:
```
[LISAv2 Test Results Summary]
[hyperv] Test Run On           : 01/16/2019 07:15:53
[hyperv] VHD Under Test        : Ubuntu_18.04.1\Ubuntu_18.04.1.vhdx
[hyperv] Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
[hyperv] Total Time (dd:hh:mm) : 0:0:6
[hyperv] 
[hyperv]    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
[hyperv] ---------------------------------------------------------------------------------------------------------------------
[hyperv]     1 BVT-NET-IFUP-IFDOWN                                                               PASS                 4.21 
[hyperv] 
```

**Comment by ader1990** [comment]
@LiliDeng can you merge this patch, if it is all good from your side? thanks

**Comment by LiliDeng** [comment]
I will do a basic check, then @lizzha can help merge

**Comment by ader1990** [comment]
because of PR #594 being merged before , I had to add the parameter to the LISAv2-Framework.psm1, so that the feature is unaffected when this patch is merged

**Pull request merged**